src/main/java/org/spongepowered/common/service/pagination/PaginationCalculator.java

```diff
@@ -85,88 +89,206 @@ public int getLinesPerPage(MessageReceiver source) {
+            int endColumn = temp & 15;
+
+            width = (endColumn + 1) - startColumn;
+            //Why does this scaling happen?
+            //I believe it makes unicode fonts skinnier to better match the character widths of the default Minecraft
+            // font however there is a int math vs float math bug in the Minecraft FontRenderer.
+            //The float math is adjusted for rendering, they attempt to do the same thing for calculating string widths
+            //using integer math, this has potential rounding errors, but we should copy it and use ints as well.
+            width = (width / 2)+1;
         } else {
-            // MC unicode -- what does this even do? but it's client-only so we can't use it directly :/
-            int j = UNICODE_CHAR_WIDTHS[codePoint] >>> 4;
-            int k = UNICODE_CHAR_WIDTHS[codePoint] & 15;
+            width = 0;
+        }
+        //if bolded width gets 1 added.
+        if(isBold && width > 0) width = width + 1;
 
-            if (k > 7) {
-                k = 15;
-                j = 0;
-            }
-            width = ((k + 1) - j) / 2 + 1;
-            if (isBold) {
-                width += 0.5;
+        return width;
+    }
+
+    /**
+     * get width using legacy methods, unsure what the difference is.
+     * @param text
+     * @return the number of character pixels/columns the line takes up
+     */
+    private int getLegacyWidth(String text){
+        if (text == null) {
+            return 0;
+        } else {
+            int total = 0;
+            boolean bold = false;
+            for (int index = 0; index < text.length(); ++index) {
+                char charElement = text.charAt(index);
+                int width = this.getWidth(charElement, bold);
+                if (charElement == TextSerializers.LEGACY_FORMATTING_CODE.getCharacter() && index < text.length() - 1) {
+                    ++index;
+                    charElement = text.charAt(index);
+
+                    if (charElement == 'l' || charElement == 'L') {
+                        bold = true;
+                    } else {
+                        if (charElement == 'r' || charElement == 'R') {
+                            bold = false;
+                        }
+                    }
+                    width = 0;
+                }
+                total += width;
             }
+            return total;
         }
-        return width;
     }
 
-    private int getLength(MessageReceiver source, Text text) {
-        double columnCount = 0d;
-        for (Text child : text.withChildren()) {
-            final String txt;
+    /**
+     * compute the length of a given text
+     * @param text
+     * @return the number of character pixels/columns the line takes up
+     */
+    @VisibleForTesting
+    int getLength(Text text) {
+        //TODO: work out what the difference is between the legacy width, and getLength, and implement it in getLength.
+        if(true)
```

Comment:
I would prefer to not just copy the Mojang code blindly, and actually understand why getLegacyWidth works more accurately then what has been effectively short circuited here. 

If you are happy with the results however I guess It's fine to just prune the pagination out.

I'm also not entirely happy with the fact that PaginationCalculator now relies upon the text mixins as it makes it harder to unit test.